### ISSUE TITLE
[WP][UX] Remove "tabs" for menu local tasks, move into contextual links (and/or admin bar)

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#464]
![zomg-it-is-not-1994](https://cloud.githubusercontent.com/assets/397895/5914559/c8785cf2-a5ac-11e4-9538-32e34147b949.png)

"Tabs" (Menu Local Tasks) are a strange UX drupalism (soon to be a backdropism too).  Before we had contextual links, tabs made some sort of sense: we needed a way for people to edit things directly from where they are viewing them.

But now we have contextual links, these links allow people to edit all kinds of things directly from where they are viewing them!  Let's move the links that are currently displayed as "tabs" into contextual links instead, and see if we can get rid of this concept of Menu Local Tasks.

We have a lot of administrative listings that are currently being displayed as tabs, for convenience. These items would need to be converted into MENU_NORMAL_ITEMS instead.  ( IMO, that's what they should have been in the first place, so this will be just a return to normalcy.)

**Comment by klonos** [comment]
Are we still pursuing this?

**Comment by quicksketch** [comment]
Not in 1.x, that's for sure. This is tagged for 2.x currently.

**Comment by klonos** [comment]
> @klonos I don't think that means admin menu needs to be required. If we move all these menu items into the normal menu system, they will appear in any menu-displaying-tool automatically, and any replacement menu system (admin module, for example) should also show them automatically.

**Comment by klonos** [comment]
To be honest, I don't like contextual menus. It's the first module I disable in Backdrop (pretty much what I used to do with the Overlay in D7). The local task tabs in the frontend are only shown to people that have proper permissions, so the "uglyness" never reached the eyes of the regular site visitor. IMO a single click to edit is better UX than requiring an extra click (to bring up the context menu). That's why I like the idea of an edit link next to the page title, but that doesn't account for the rest of the menu items. It also might be confusing because people might think that the edit link will allow them to edit the page title alone - not the content itself (especially for people that have seen/used this pattern in other places like the machine name edit link and GitHub issue titles for example).

**Comment by klonos** [comment]
FTR: WordPress uses an "Edit page" item (with a pencil icon) in their admin bar:

![wp-edit_page](https://cloud.githubusercontent.com/assets/2423362/22167990/dd0da238-dfbd-11e6-8cd9-6f3f9fbf95b3.jpg)

**Comment by jenlampton** [comment]
Admin bar ++

**Comment by klonos** [comment]
> Admin bar ++

Yep, so we cycled back to [what I've said back in 2014](https://github.com/backdrop/backdrop-issues/issues/464#issuecomment-67468472):

> So, that makes the admin bar a required module in Backdrop. Right? Either that or any alternatives will need to supply the same links we move to it.

**Comment by klonos** [comment]
...I guess that when local menus were introduced people have thought of all these alternatives (admin bar, contextual links) and saw the need to have them be required modules (otherwise there was a risk of "locking" users outside of being able to edit their content).
